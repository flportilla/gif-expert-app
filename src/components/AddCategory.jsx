import PropTypes from 'prop-types';
import React, { useState } from 'react'

export const AddCategory = ({ onNewvalue }) => {

    const [inputValuie, setinputValuie] = useState('')

    const handleInputChange = ({ target }) => {
        setinputValuie(target.value)
    }

    const handleSubmit = (event) => {
        event.preventDefault()
        const newValue = inputValuie.trim()
        if (newValue.length <= 1) return;

        onNewvalue(newValue)
        setinputValuie('')
    }

    return (
        <form onSubmit={handleSubmit} aria-label="form">
            <input
                type="text"
                value={inputValuie}
                onChange={handleInputChange}
                placeholder="search gifs"
            />
        </form>
    )
}

AddCategory.propTypes = {
    onNewvalue: PropTypes.func.isRequired,
}
