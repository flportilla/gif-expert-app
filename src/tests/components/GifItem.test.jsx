import { render, screen } from '@testing-library/react'
import { GifItem } from '../../components/GifItem'

describe('Tests on <GiftItem />', () => {

    const title = 'Alucard'
    const url = 'https://castlevania.com/random.jpg'

    test('Must match the snapshot', () => {

        const { container } = render(<GifItem title={title} url={url} />);
        expect(container).toMatchSnapshot();

    })

    test('should show the img with the URL and ALT indicated', () => {

        render(<GifItem title={title} url={url} />);

        const { src, alt } = screen.getByRole('img');

        expect(src).toBe(url);
        expect(alt).toBe(title);

    })

    test('should show the title on the component', () => {

        render(<GifItem title={title} url={url} />);
        expect(screen.getByText(title)).toBeTruthy();

    })
})