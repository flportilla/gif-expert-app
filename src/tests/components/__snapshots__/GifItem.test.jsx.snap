// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <GiftItem /> Must match the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Alucard"
      src="https://castlevania.com/random.jpg"
    />
    <p>
      Alucard
    </p>
  </div>
</div>
`;
